E402 doesn't respect version strings

``` python
# some header comment 1
# some header comment 2
# some header comment 3
"DocString"

__version__ = "3.1"

import os


def main():
    print os.getenv('NOTREAL', 0)

if __name__ == '__main__':
    main()
```

pep8 returns
testblah.py:8:1: E402 module level import not at top of file

When this is acceptable by PEP8's standard on version strings.

https://www.python.org/dev/peps/pep-0008/#version-bookkeeping

pep8 --version
1.6.2

--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -134,6 +134,7 @@ STARTSWITH_INDENT_STATEMENT_REGEX = re.compile(
         'while',
     )))
 )
+DUNDER_REGEX = re.compile(r'^__([^\s]+)__ = ')
 
 # Work around Python < 2.6 behaviour, which does not generate NL after
 # a comment which is on a line by itself.
@@ -955,6 +956,8 @@ def module_imports_on_top_of_file(
     if line.startswith('import ') or line.startswith('from '):
         if checker_state.get('seen_non_imports', False):
             yield 0, "E402 module level import not at top of file"
+    elif re.match(DUNDER_REGEX, line):
+        return
     elif any(line.startswith(kw) for kw in allowed_try_keywords):
         # Allow try, except, else, finally keywords intermixed with imports in
         # order to support conditional importing

--- a/testsuite/E40.py
+++ b/testsuite/E40.py
@@ -11,8 +11,16 @@ from foo.bar.yourclass import YourClass
 
 import myclass
 import foo.bar.yourclass
-#: E402
+#: Okay
 __all__ = ['abc']
+
+import foo
+#: Okay
+__version__ = "42"
+
+import foo
+#: Okay
+__author__ = "Simon Gomizelj"
 
 import foo
 #: Okay

